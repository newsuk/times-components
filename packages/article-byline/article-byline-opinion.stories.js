import { showcaseConverter } from "@times-components/storybook";
import showcase from "./article-byline-opinion.showcase";

showcaseConverter(module, showcase);
