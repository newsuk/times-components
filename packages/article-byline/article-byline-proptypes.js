import PropTypes from "prop-types";
import { MarkupBuilder } from "@times-components/markup";
import { Text } from "react-native";

const articleBylinePropTypes = {
  ast: MarkupBuilder.propTypes.ast,
  style: PropTypes.shape({
    byline: Text.propTypes.style,
    link: Text.propTypes.style
  })
};

export default articleBylinePropTypes;
