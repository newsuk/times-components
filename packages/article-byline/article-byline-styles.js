export const nativeStyles = {
  byline: {
    color: "#696969",
    fontSize: 13,
    fontFamily: "GillSansMTStd-Medium"
  },
  link: {
    color: "#069"
  }
};

export const webStyles = {
  ...nativeStyles
};

webStyles.link.textDecorationLine = "none";
