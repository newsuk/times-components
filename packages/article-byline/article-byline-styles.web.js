const styles = {
  byline: {
    fontSize: 13,
    fontFamily: "GillSansMTStd-Medium",
    whiteSpace: "pre"
  },
  bylineColor: {
    color: "#696969"
  },
  container: {
    flexDirection: "row"
  }
};

export default styles;
