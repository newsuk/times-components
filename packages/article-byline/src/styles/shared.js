import styleguide from "@times-components/styleguide";

const { colours, fontFactory } = styleguide();
const shared = {
  link: {
    color: colours.functional.action,
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    })
  },
  text: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    flexDirection: "row"
  }
};

export default shared;
