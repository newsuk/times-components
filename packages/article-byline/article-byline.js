/* eslint-disable react/no-array-index-key */
import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { TextLink } from "@times-components/link";
import { renderTrees } from "@times-components/markup";
import styles from "./article-byline-styles";

import {
  articleBylinePropTypes,
  articleBylineDefaultPropTypes
} from "./article-byline-proptypes";

const linkStyles = StyleSheet.create({
  link: {
    color: "#069"
  }
});

const ArticleByline = ({ ast, style }) => (
  <View style={[styles.container, style.container]}>
    <Text
      accessibilityLabel="articleByline"
      testID="articleByline"
      style={[styles.byline, style.byline, styles.bylineColor]}
    >
      {renderTrees(ast, {
        author(key, attributes, children) {
          const url = `/profile/${attributes.slug}`;
          return (
            <TextLink
              style={[linkStyles.link, style.link]}
              key={key}
              url={url}
              onPress={() => {}}
            >
              {children}
            </TextLink>
          );
        }
      })}
    </Text>
  </View>
);

export default ArticleByline;

ArticleByline.propTypes = articleBylinePropTypes;

ArticleByline.defaultProps = articleBylineDefaultPropTypes;
