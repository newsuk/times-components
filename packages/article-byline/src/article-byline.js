import React from "react";
import { Text } from "react-native";
import renderByline from "./render-byline";
import { propTypes, defaultProps } from "./article-byline-prop-types";
import styles from "./styles";

const renderAuthorComponent = (
  children,
  key,
  attributes,
  { className, bylineStyle }
) => (
  <Text
    className={className}
    key={key}
    style={[styles.nonLinkText, bylineStyle]}
  >
    {children}
  </Text>
);

const ArticleByline = ({ ast, ...props }) =>
  renderByline(renderAuthorComponent, ast, styles.nonLinkText, props);

ArticleByline.displayName = "ArticleByline";

ArticleByline.propTypes = propTypes;
ArticleByline.defaultProps = defaultProps;

export { default as ArticleBylineOpinion } from "./article-byline-opinion";
export { default as ArticleBylineWithLinks } from "./article-byline-with-links";
export { default as hasBylineData } from "./has-byline-data";
export default ArticleByline;
