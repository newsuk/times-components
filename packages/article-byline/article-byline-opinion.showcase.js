/* eslint-disable react/prop-types */
import React from "react";
import { View } from "react-native";
import { ArticleBylineOpinion } from "./src/article-byline";

const authorsAST = require("./fixtures/authors.json");

const preventDefaultedAction = decorateAction =>
  decorateAction([
    ([e, ...args]) => {
      e.preventDefault();
      return ["[SyntheticEvent (storybook prevented default)]", ...args];
    }
  ]);

const ComponentWrapper = ({ children }) => (
  <View style={{ flexDirection: "row", flexWrap: "wrap" }}>{children}</View>
);

const getProps = decorateAction => ({
  onAuthorPress: preventDefaultedAction(decorateAction)("onAuthorPress")
});

export default {
  children: [
    {
      component: (_, { decorateAction }) => (
        <ComponentWrapper>
          <ArticleBylineOpinion
            ast={authorsAST.singleAuthor}
            {...getProps(decorateAction)}
          />
        </ComponentWrapper>
      ),
      name: "Byline with a single author",
      type: "story"
    }
  ],
  name: "Primitives/Article Byline/Opinion"
};
