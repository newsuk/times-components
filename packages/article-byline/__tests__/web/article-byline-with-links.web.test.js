import { ArticleBylineWithLinks } from "../../src/article-byline";
import shared from "../shared";

shared(ArticleBylineWithLinks);
