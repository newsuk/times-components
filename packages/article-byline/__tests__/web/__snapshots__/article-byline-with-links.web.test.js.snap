// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. single inline element 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
    Lindsay McIntosh, Scottish Political Editor
  </div>
</div>
`;

exports[`2. with the author in the begining 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/camilla-long"
    role="link"
  >
    Camilla Long
  </a>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
    , Environment Editor
  </div>
</div>
`;

exports[`3. with the author at the end 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
    RBS Six Nations, 
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/mike-atherton"
    role="link"
  >
    Mike Atherton
  </a>
</div>
`;

exports[`4. with multiple authors separated by a pipe 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/camilla-long"
    role="link"
  >
    Camilla Long
  </a>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
     | 
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/mike-atherton"
    role="link"
  >
    Mike Atherton
  </a>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
     | 
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/alexandra-frean"
    role="link"
  >
    Alexandra Frean
  </a>
</div>
`;

exports[`5. with multiple authors separated by text with commas 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/camilla-long"
    role="link"
  >
    Camilla Long
  </a>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
    , 
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/mike-atherton"
    role="link"
  >
    Mike Atherton
  </a>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
    , 
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/alexandra-frean"
    role="link"
  >
    Alexandra Frean
  </a>
</div>
`;

exports[`6. with multiple authors separated by spaces 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/camilla-long"
    role="link"
  >
    Camilla Long
  </a>
  <div
    className="css-text-901oao r-color-1khp51w r-flexDirection-18u37iz r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n"
  >
     
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/mike-atherton"
    role="link"
  >
    Mike Atherton
  </a>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao r-color-j7kcqb r-cursor-1loqt21 r-fontFamily-j2s0nr r-fontSize-n6v787 r-lineHeight-fxxt2n r-textDecorationLine-1ddef8g"
    href="/profile/alexandra-frean"
    role="link"
  >
    Alexandra Frean
  </a>
</div>
`;

exports[`7. null with an empty ast 1`] = `null`;
